### PR TITLE
Fix panic 'provided file is not a console'

### LINKED ISSUE #3433: Panic: `not a console`, follow-up to #3297
### Description

As outlined in #3297, there are more places that are likely to panic when we do not have a tty.

While the patch in #3300 fixed it for `exec` for a specific use case, clearly `run` can crash in the same circumstances.

Suggestion is to wrap the third party term console library so that it returns an error we can check instead of crashing.

### Steps to reproduce the issue

na

### Describe the results you received and expected

na

### What version of nerdctl are you using?

main

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

None

### Host information

_No response_

### PR DESCRIPTION
**Pull request by apostasie**
See commit message.

Fixes #3433

This should put this panic issue to rest.

### CODE REVIEW

**Comment by apostasie** [comment]
Ping @AkihiroSuda - pending green CI.

**Comment by apostasie** [comment]
Ahhhh windows, windows.

@sondavidb @jsturtevant any idea for me on this?

**Comment by sondavidb** [comment]
Was trying to set up containerd on my local machine and it has been a lot more difficult than I anticipated 😓

Looks like the change you made had the integration pass but not the canary. Is this test known to be flaky? Looking through the changes it seems fine so I'm pretty stumped as well

**Comment by apostasie** [comment]
@sondavidb I think `TestRunWithTtyAndDetached` is failing very often - but may succeed occasionally.

Might be a recent regression, and unrelated to this changeset, as my other PRs get the failure too.

Investigating...

**Comment by apostasie** [comment]
@sondavidb maybe a side-effect of https://github.com/containerd/nerdctl/pull/3304 which did move that test around?

If there is some racyness involved, moving it in a different place may produce more adverse timing conditions?

Anyhow.

Ping @AkihiroSuda @jsturtevant as it seems to fail very often.

**Comment by apostasie** [comment]
Test is broken on main.

Moving discussion to #3437

**Comment by apostasie** [comment]
@AkihiroSuda besides what appears to be an unrelated regression / flakyness (#3437) this is ready.

File: pkg/consoleutil/consoleutil.go [hunk 1]
```diff
@@ -18,8 +18,22 @@ package consoleutil
 
 import (
 	"context"
+	"os"
+
+	"github.com/containerd/console"
 )
 
+// Current is from https://github.com/containerd/console/blob/main/console.go#L68-L80
```
Comment (AkihiroSuda) [inline pkg/consoleutil/consoleutil.go:26]: s/main/<specific version>/

File: cmd/nerdctl/container/container_exec.go [hunk 2]
```diff
@@ -88,13 +86,6 @@ func processExecCommandOptions(cmd *cobra.Command) (types.ContainerExecOptions,
 		}
 	}
 
-	_, isTerminal := term.GetFdInfo(os.Stdin)
-	if !flagD {
-		if flagT && flagI && !isTerminal {
-			return types.ContainerExecOptions{}, errors.New("the input device is not a TTY")
-		}
-	}
-
```
Comment (AkihiroSuda) [inline cmd/nerdctl/container/container_exec.go:97]: Maybe add a comment line to explain why there is no need to check `isTerminal`

Comment (apostasie) [reply cmd/nerdctl/container/container_exec.go:97]: Done

File: pkg/consoleutil/consoleutil.go (hunk 1 above)
Comment (apostasie) [reply pkg/consoleutil/consoleutil.go]: Done.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**